### ISSUE TITLE
Suggestion for Improving Training Checkpoint Recovery Functionality

### ISSUE THREAD

**Issue by yiyi0209** [ZhaochongAn/COSeg#21]
Hello!

While reproducing the code, I noticed that the current checkpoint implementation only saves the model parameters (state_dict) without preserving training state information (such as optimizer and learning rate scheduler states). This makes it impossible to fully resume training after interruptions (requiring restarting from scratch).

Original code in main_fs.py, line 567:
torch.save({"state_dict": model.state_dict()}, filename)

Suggested improvement:
torch.save({
    'epoch': epoch + 1,
    'state_dict': model.state_dict(),
    'optimizer': optimizer.state_dict(),
    'scheduler': scheduler.state_dict(),
    'best_iou': best_iou
}, "model_last.pth")

Benefits:
· Enables complete recovery from checkpoints (epoch, optimizer, learning rate, etc.)
· Maintains backward compatibility (old checkpoints can still be loaded, new fields are automatically ignored)
· No performance impact

Thank you for your valuable open-source contribution!

**Comment by YunZhou0321** [comment]
That's a great modification!
